cms/models/pagemodel.py

```diff
@@ -1095,19 +1128,24 @@ def get_absolute_url(self, language=None, fallback=True):
 
         with force_language(language):
             try:
-                if self.path == '':
-                    return reverse('pages-root')
-                return reverse('pages-details-by-slug', kwargs={"slug": self.path})
+                if self.path == "":
+                    return reverse("pages-root")
+                return reverse("pages-details-by-slug", kwargs={"slug": self.path})
             except NoReverseMatch:
                 return None
 
-    def get_path_for_base(self, base_path=''):
-        old_base, sep, slug = self.path.rpartition('/')
-        return f'{base_path}/{slug}' if base_path else slug
+    def get_path_for_base(self, base_path=""):
+        old_base, sep, slug = self.path.rpartition("/")
+        return f"{base_path}/{slug}" if base_path else slug
 
+    def save(self, *args, **kwargs):
```

Comment:
Why did you add this? Did we have instances where the `site` property was not set?

Reply:
I denormalized the site above in line 1102, so I'm saving the site to this model from its page attribute so I can use it as a field in our `UniqueConstraint`.